### PR TITLE
Add testIndexLatest

### PR DESCRIPTION
**Pull request by DaveCTurner**
Add a third-party repo test to verify that we do actually write a blob
called `index.latest` when writing to the repository. As described in
the code comment, this is because ESS uses the freshness of this blob to
protect against incorrect repository cleanups, so we want to make sure
we don't stop writing it in future.

### CODE REVIEW

**Review by fcofdez — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-4

File: test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java [hunk 1]
```diff
@@ -218,6 +224,58 @@ public void testCleanup() throws Exception {
...
+        final var createSnapshot2Response = clusterAdmin().prepareCreateSnapshot(TEST_REPO_NAME, randomIdentifier())
+            .setWaitForCompletion(true)
+            .get();
+        assertTrue(blobContents.add(readIndexLatest(repository)));
+
+        assertAcked(clusterAdmin().prepareDeleteSnapshot(TEST_REPO_NAME, createSnapshot1Response.getSnapshotInfo().snapshotId().getName()));
+        assertTrue(blobContents.add(readIndexLatest(repository)));
+
+        assertAcked(clusterAdmin().prepareDeleteSnapshot(TEST_REPO_NAME, createSnapshot2Response.getSnapshotInfo().snapshotId().getName()));
+        assertTrue(blobContents.add(readIndexLatest(repository)));
```
Comment (ywangd) [inline test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java:259]: Is it useful to have a cleanup call as well?

**Review by ywangd — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java (hunk 1 above)
Comment (DaveCTurner) [reply test/framework/src/main/java/org/elasticsearch/repositories/AbstractThirdPartyRepositoryTestCase.java:259]: I worry that we might make repo cleanup detect no-op runs better in future which would invalidate the assertion in this test that the `index.latest` blob changes. I think snapshot creation & deletion is enough.

**Pull request merged**